demo/api/src/products/product-category.resolver.ts

```diff
@@ -0,0 +1,60 @@
+import { AffectedEntity, CometValidationException, RequiredPermission } from "@comet/cms-api";
+import { InjectRepository } from "@mikro-orm/nestjs";
+import { EntityManager, EntityRepository } from "@mikro-orm/postgresql";
+import { Args, ID, Int, Mutation, Resolver } from "@nestjs/graphql";
+
+import { ProductCategory } from "./entities/product-category.entity";
+import { ProductCategoriesService } from "./generated/product-categories.service";
+
+@Resolver(() => ProductCategory)
+@RequiredPermission(["products"], { skipScopeCheck: true })
+export class ProductCategoryResolver {
+    constructor(
+        private readonly entityManager: EntityManager,
+        private readonly productCategoriesService: ProductCategoriesService,
+        @InjectRepository(ProductCategory) private readonly repository: EntityRepository<ProductCategory>,
+    ) {}
+
+    @Mutation(() => Boolean)
+    @AffectedEntity(ProductCategory)
+    async updateProductCategoryPosition(
+        @Args("id", { type: () => ID }) id: string,
+        @Args("position", { type: () => Int }) newPosition: number,
+    ): Promise<boolean> {
+        if (newPosition < 1) {
+            throw new CometValidationException('"position" cannot be smaller than 1');
+        }
+
+        const productCategory = await this.repository.findOneOrFail(id);
+        const { position: oldPosition } = productCategory;
+
+        const count = await this.repository.count({}); // add filter for grouping if necessary
+
+        if (newPosition > count) {
+            throw new CometValidationException(`"position" cannot be greater than ${count}`);
+        }
+
+        await this.entityManager.transactional(async (em) => {
```

Comment:
I think this code could also be placed in product-category.service.ts. I placed it in this resolver because how easy it is to combine it with the generated api code.